config/src/test/scala/com/typesafe/config/impl/ConfigBeanFactoryTest.scala

```diff
@@ -71,6 +73,14 @@ class ConfigBeanFactoryTest extends TestUtils {
     }
 
     @Test
+    def testCreateEnum() {
+        val beanConfig: EnumsConfig = ConfigBeanFactory.create(loadConfig().getConfig("enums"), classOf[EnumsConfig])
+        assertNotNull(beanConfig)
+        assertEquals(Problem.P1, beanConfig.getProblem)
+        assertEquals(ArrayBuffer(Solution.S1, Solution.S3), beanConfig.getSolutions.asScala)
+    }
+
+    @Test
```

Comment:
it'd be good to test the error case too (invalid value for the enum). There are some error tests up above for other types.